src/tracing/json/json_tracing.cpp

```diff
@@ -56,20 +56,89 @@ using namespace chip::Decoders;
 
 using PayloadDecoderType = chip::Decoders::PayloadDecoder<64, 256>;
 
+/// Figures out a unique name within a json object.
+///
+/// Decoded keys may be duplicated, like list elements are denoted as "[]".
+/// The existing code does not attempt to encode lists and everything is an object,
+/// so this name builder attempts to find unique keys for elements inside a json.
+///
+/// In particular a repeated "[]", "[]", ... will become "[0]", "[1]", ...
+class UniqueNameBuilder
+{
+public:
+    UniqueNameBuilder(chip::StringBuilderBase & formatter) : mFormatter(formatter) {}
+    const char * c_str() const { return mFormatter.c_str(); }
+
+    // Figure out the next unique name in the given value
+    //
+    // After this returns, c_str() will return a name based on `baseName` that is
+    // not a key of `value` (unless on overflows, which are just logged)
+    void ComputeNextUniqueName(const char * baseName, ::Json::Value & value)
+    {
+        FirstName(baseName);
+        while (value.isMember(mFormatter.c_str()))
+        {
+            NextName(baseName);
+            if (!mFormatter.Fit())
+            {
+                ChipLogError(Automation, "Potential data loss: insufficient space for unique keys in json");
+                return;
+            }
+        }
+    }
+
+private:
+    void FirstName(const char * baseName)
+    {
+        if (strcmp(baseName, "[]") == 0)
```

Comment:
Suggest replacing anonymous tag with something that doesn't look like list indexing.

Reply:
Updated to `Anonymous<>` (and injected index in there)